.github/workflows/update-maintainers.yml

```diff
@@ -0,0 +1,179 @@
+name: Update maintainers list
+
+on:
+  push:
+    branches: ["main"]
+    paths: ["lib/maintainers.nix"]
+  schedule:
+    # Update every Monday at 9 AM UTC
+    - cron: "0 9 * * 1"
+  workflow_dispatch:
+    inputs:
+      create_pr:
+        description: "Create PR even if no changes"
+        required: false
+        default: false
+        type: boolean
+
+jobs:
+  update-maintainers:
+    runs-on: ubuntu-latest
+    # Creating PRs to update the generated maintainers file
+    permissions:
+      contents: write
+      pull-requests: write
+    if: github.repository_owner == 'nix-community' || github.event_name == 'workflow_dispatch'
+    steps:
+      - name: Create GitHub App token
+        uses: actions/create-github-app-token@v2
+        if: vars.CI_APP_ID
+        id: app-token
+        with:
+          app-id: ${{ vars.CI_APP_ID }}
+          private-key: ${{ secrets.CI_APP_PRIVATE_KEY }}
+          permission-contents: write
+          permission-pull-requests: write
+
+      - name: Get GitHub App user info
+        id: user-info
+        if: vars.CI_APP_ID
+        env:
+          GH_TOKEN: ${{ steps.app-token.outputs.token }}
+          slug: ${{ steps.app-token.outputs.app-slug }}
```

Comment:
Here, we could do something like:

```yaml
- name: Get GitHub App user info
  id: user-info
  env:
    GH_TOKEN: ${{ steps.app-token.outputs.token || github.token }}
    slug: ${{ steps.app-token.outputs.app-slug || 'github-actions' }}
    if: ${{ vars.CI_APP_ID && '' || '41898282' }}
  run: |
    name="$slug[bot]"
    if [ -z "$id" ]; then
      id=$(gh api "/users/$name" --jq .id)
    fi

    # ...
```

Reply:
> Here, we could do something like:

Or it might be simpler to handle that in the `setup-git` step's `env` inputs

Reply:
> I'm guessing at some point we'll want to extract this "gen token and setup git" logic into a composite action?

Was thinking about that too while bouncing around all these workflows.